### PR TITLE
Add language selector and session language persistance

### PR DESCRIPTION
**Pull request by FrancoARossi**
This pull request introduces a language selector to the footer of the web UI. Additionally, it ensures that the selected locale persists throughout the current session. If the user doesn't choose a language, it will default to their preferred one.

### Screenshots

**Desktop**
Screenshot 2024-03-08 at 19 56 02
Screenshot 2024-03-08 at 19 56 18
Screenshot 2024-03-08 at 19 56 33

**Mobile**
Screenshot 2024-03-08 at 19 57 46
Screenshot 2024-03-08 at 19 57 58

### CODE REVIEW

**Comment by mperham** [comment]
But why? What’s the usecase?

**Comment by FrancoARossi** [comment]
> But why? What’s the usecase?

I feel that enforcing the language based on the browser is somewhat restrictive. I though of doing this because a couple of days ago I was screen sharing to some co-workers from the US (me being from Argentina) to debug some failing jobs and it was a little awkward for them to see the UI in spanish without having the option to change it, I also would like to have it set to english as it's the language that I use daily while coding...

File: web/views/_footer.erb [hunk 1]
```diff
@@ -15,7 +15,18 @@
             <p class="navbar-text"><a rel=help href="https://github.com/sidekiq/sidekiq/wiki">docs</a></p>
           </li>
           <li>
-            <p class="navbar-text"><a rel=external href="https://github.com/sidekiq/sidekiq/tree/main/web/locales"><%= locale %></a></p>
+            <form id="locale-form" action="<%= root_path %>change_locale" method="post">
+              <%= csrf_tag %>
+              <select id="locale-select" class="locale-select" name="locale">
+                <% available_locales.each do |locale_option| %>
+                  <% if locale_option === locale %>
```
Comment (mperham) [inline web/views/_footer.erb:22]: Be careful. By using triple ===, I believe you may be risking a security hole, allowing anyone to inject any value for the `locale` parameter in the session. Thankfully you aren't rendering it.

File: web/assets/stylesheets/application-rtl.css [hunk 2]
```diff
@@ -151,3 +151,25 @@ div.interval-slider {
     padding-left: 5px;
   }
 }
+
+.locale-select {
```
Comment (mperham) [inline web/assets/stylesheets/application-rtl.css:155]: Can you DRY this up, especially since many rules overlap with application.css?

**Comment by mperham** [comment]
Remove all of the explicit styling and add the bootstrap form styles. The dropdown should look like the other dropdown in the UI (the metrics time selector).

**Comment by FrancoARossi** [comment]
> Remove all of the explicit styling and add the bootstrap form styles. The dropdown should look like the other dropdown in the UI (the metrics time selector).

I've applied the `.form-inline` and `.form-control` classes from Bootstrap, and I've also kept the bare minimum to align it properly with the other footer elements.

**Comment by mperham** [comment]
The only obvious remaining issue is lack of dark mode support, we need a dark background for that form element. Our old Bootstrap CSS predates dark mode so I'm not sure if this can activate automatically. I've put in #6222 to upgrade to a modern Bootstrap.

File: web/views/_footer.erb [hunk 3]
```diff
@@ -15,7 +15,19 @@
             <p class="navbar-text"><a rel=help href="https://github.com/sidekiq/sidekiq/wiki">docs</a></p>
           </li>
           <li>
-            <p class="navbar-text"><a rel=external href="https://github.com/sidekiq/sidekiq/tree/main/web/locales"><%= locale %></a></p>
+            <form id="locale-form" class="form-inline" action="<%= root_path %>change_locale" method="post">
+              <%= csrf_tag %>
+              <label class="sr-only" for="locale">Language</label>
```
Comment (mperham) [inline web/views/_footer.erb:20]: Good work on this accessibility feature!

**Comment by tierra** [comment]
In my instance, our footer is already quite wide, kicking this selector to the next line. I believe because of that, the footer is now larger height, causing some page content to be hidden behind the footer. This is especially problematic on pages with buttons on the bottom.

Note the slight window width changes, with both these being fully scrolled to bottom on the same page:

image

Slightly shorter, triggering the width break, and hiding the buttons:

image

**Pull request merged**